### PR TITLE
feat: wait for document states

### PR DESCRIPTION
**Pull request by aaarichter**
* browser.waitForDocumentReady()
* browser.waitForLoadEvent()

### CODE REVIEW

File: lib/browser/navigation.js [hunk 1]
```diff
@@ -70,6 +70,28 @@ function urlAsserter(url, query) {
...
+  timeout = timeout || 8000;
+  var expr;
+  if (state === 'interactive') {
+    expr = "var state; (state = document.readyState) === 'interactive' || state === 'complete';";
+  }
+  if (state === 'complete') {
+    expr = "document.readyState === 'complete'";
+  }
+  self.setAsyncScriptTimeout(timeout + 250);
+  return self.waitForConditionInBrowser(expr || '', timeout);
```
Comment (jkrems) [inline lib/browser/navigation.js]: Should this fail loudly instead of defaulting to an empty expression?

File: lib/browser/navigation.js [hunk 2]
```diff
@@ -70,6 +70,28 @@ function urlAsserter(url, query) {
...
+ *
+ * @param {object} self
+ * @param {string} state
+ * @param {number} [timeout] - Timeout for js condition to complete (optional)
+ */
+function waitForDocumentState(self, state, timeout) {
+  timeout = timeout || 8000;
+  var expr;
+  if (state === 'interactive') {
+    expr = "var state; (state = document.readyState) === 'interactive' || state === 'complete';";
```
Comment (jkrems) [inline lib/browser/navigation.js]: Is reading `document.readyState` twice really expensive? Or why is this assigning to a variable?

File: lib/browser/navigation.js [hunk 3]
```diff
@@ -70,6 +70,28 @@ function urlAsserter(url, query) {
...
+ * https://developer.mozilla.org/en-US/docs/Web/API/Document/readyState
+ *
+ * @param {object} self
+ * @param {string} state
+ * @param {number} [timeout] - Timeout for js condition to complete (optional)
+ */
+function waitForDocumentState(self, state, timeout) {
+  timeout = timeout || 8000;
+  var expr;
+  if (state === 'interactive') {
```
Comment (jkrems) [inline lib/browser/navigation.js]: Might be a good candidate for a switch statement (that would also be a good opportunity to `throw` in the `default:` branch).

File: lib/browser/navigation.js (hunk 2 above)
Comment (aaarichter) [reply lib/browser/navigation.js]: no it isn't expensive.

**Comment by aaarichter** [comment]
feedback addressed @jkrems

**Pull request merged**